pkg/libs/endpointaccessible/endpoint_accessible_controller_test.go

```diff
@@ -48,12 +49,27 @@ func Test_endpointAccessibleController_sync(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "endpoint check disabled",
+			endpointCheckDisabledFunc: func() (bool, error) {
+				return true, nil
+			},
+			wantErr: false,
+		},
+		{
+			name: "endpoint check disabled func returns error",
+			endpointCheckDisabledFunc: func() (bool, error) {
+				return false, fmt.Errorf("fake error")
+			},
+			wantErr: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			c := &endpointAccessibleController{
-				operatorClient: v1helpers.NewFakeOperatorClient(&operatorv1.OperatorSpec{}, &operatorv1.OperatorStatus{}, nil),
-				endpointListFn: tt.endpointListFn,
+				operatorClient:            v1helpers.NewFakeOperatorClient(&operatorv1.OperatorSpec{}, &operatorv1.OperatorStatus{}, nil),
+				endpointListFn:            tt.endpointListFn,
+				endpointCheckDisabledFunc: tt.endpointCheckDisabledFunc,
```

Comment:
Should we add a check to this test that if this returns true that the controller conditions are properly removed?